ashley/src/main/kotlin/ktx/ashley/engine/engines.kt

```diff
@@ -0,0 +1,52 @@
+package ktx.ashley.engine
+
+import com.badlogic.ashley.core.Component
+import com.badlogic.ashley.core.Engine
+import com.badlogic.ashley.core.Entity
+import ktx.ashley.AshleyDsl
+
+/**
+ * Creates an instance of the component [T] and adds it to this [Entity].
+ *
+ * @param T the [Component] type to get or create
+ * @param configure inlined function with [T] as the receiver to allow additional configuration of the [Component]
+ * @return the created [Component]
+ */
+inline fun <reified T: Component> Entity.with(configure: (@AshleyDsl T).() -> Unit): T {
+  return with<T>().also(configure)
+}
+
+/**
+ * Creates a pooled instance of the component [T] and adds it to this [Entity].
+ *
+ * @param T the [Component] type to get or create
+ * @return the pooled [Component]
+ */
+inline fun <reified T: Component> Entity.with(): T {
+  val component = T::class.java.newInstance()
```

Comment:
I normally use `PooledEngine`, and to be honest added the `Engine` implementation as an afterthought as it seemed more complete to provide parity for that API.

An alternative is to let the non-pooled DSL diverge so that:
```kotlin
entity<Transform> {
  x =1f
  y =2f
}
```
becomes:
```kotlin
entity(Transform()) {
  x =1f
  y =2f
}
```

Potentially we could just remove the version for non-pooled `Engine`s - this would prevent the ambiguity you mentioned by importing the wrong DSL package.

Do you have a preference? I'm swaying to just removing the `Engine` DSL if we don't think people don't need it.